seisbench/data/inspection.py

```diff
@@ -0,0 +1,410 @@
+from __future__ import annotations
+
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import TYPE_CHECKING, NamedTuple
+
+import numpy as np
+import pandas as pd
+from pyrocko.model.station import Station
+from tqdm import tqdm
+
+import seisbench
+from seisbench.data.base import WaveformDataset
+
+try:
+    from pyrocko.gui.marker import PhaseMarker, save_markers
+    from pyrocko.io import save
+    from pyrocko.model import Event, Station, dump_events, dump_stations_yaml
+    from pyrocko.trace import NoData, Trace, degapper, snuffle
+except ImportError as e:
+    raise ImportError("pyrocko is required for dataset inspection") from e
+
+if TYPE_CHECKING:
+    from seisbench.data.base import TraceParameters
+
+
+_PYROCKO_POLARITY_MAP = {
+    "undecideable": None,
+    "up": 1,
+    "down": -1,
+}
+
+logger = seisbench.logger
+
+
+class DatasetInspection:
+    _dataset: WaveformDataset
+    _metadata: pd.DataFrame
+
+    def __init__(self, dataset: WaveformDataset) -> None:
+        self._dataset = dataset
+        self._metadata = dataset.metadata
+
+    def get_pyrocko_traces(self, sample_idx: int) -> list[Trace]:
+        """
+        Returns the waveforms of a sample as a list of pyrocko Trace objects.
+
+        Requires pyrocko to be installed.
+
+        :param idx: Idx of sample to return traces for
+        :return: List of pyrocko Trace objects
+        """
+
+        ds = self._dataset
+
+        data, metadata = ds.get_sample(sample_idx)
+        traces = []
+
+        location_code = metadata.get("station_location_code")
+        location_code = "" if np.isnan(location_code) else location_code
```

Comment:
This will fail if the location code is not numerical.

Reply:
Indeed, location codes in SEED are strings. 0-2 alphanumeric uppercase (or dash "-"!) characters today, up to eight characters in future. Note "01" != "1", "00" != "". https://docs.fdsn.org/projects/source-identifiers/en/latest/definition.html